sktime/forecasting/compose/_autoregressive.py

```diff
@@ -0,0 +1,303 @@
+        )
+
+        super().__init__()
+
+    def _fit(self, y, X=None, fh=None):
+        # ignore the fh provided fit
+        _fh = ForecastingHorizon(range(1, self.horizon_length + 1))
+
+        self.forecaster.fit(y=y, X=X, fh=_fh)
+
+    def _predict(self, fh, X=None, y=None):
+        # use fh to find the maximum length to forecast regressively
+        max_fh = max(
+            *fh.to_relative(self._cutoff)._values,
+            self.horizon_length,
+        )
+
+        hist_y = self._y if y is None else y
+        hist_y = deepcopy(hist_y)  # make copy because we are going to add values to it
+
+        # calculate initial length of hist_y
+        if not isinstance(hist_y.index, pd.MultiIndex):
+            initial_length = len(hist_y.index)
+        else:
+            initial_length = len(hist_y.index.levels[-1])
+
+        if y is None and X is not None:
+            # special case of non global forecasting
+            # here X contains only (future) values
+            # however for the interface, X should contain (hist + future) values
+            # since this is non-global forecasting, we can use hist from self._X
+            X = pd.concat([self._X, X])
+
+        for i in range(max_fh - self.horizon_length + 1):
+            # truncate to the needed lengths
+            _y = self._truncate(hist_y, end_index=initial_length + i)
+            _x = self._truncate(X, end_index=initial_length + i + self.horizon_length)
+
+            # forecast for self.horizon_length
+            preds = self.forecaster.predict(y=_y, X=_x)
+
+            # use predicted values as history for next iteration
+            hist_y = self._concate_preds(hist_y, preds)
+
+        # collect forecasting points from hist_y
+        # code works for all indexes
+        absolute_horizons = fh.to_absolute_index(self.cutoff)
+        dateindex = hist_y.index.get_level_values(-1).map(
+            lambda x: x in absolute_horizons
+        )
+        preds = hist_y.loc[dateindex]
+
+        return preds
+
+    def _truncate(self, data, end_index):
+        # keep rows till this index in data
+
+        if data is None:
+            # when there is no X (exogenous data)
+            return None
+
+        if not isinstance(data.index, pd.MultiIndex):
+            return data.head(end_index)
+
+        # truncating for Multi-Index
+        # levels other than the timestamps (innermost/base level)
+        groupby_levels = list(range(len(data.index.names) - 1))
+        data = data.groupby(
+            level=groupby_levels,
+            group_keys=True,  # to keep stable conversion with pandas<2.0.0
+        ).apply(lambda x: x.head(end_index))
+        data.index = data.index.droplevel(groupby_levels)
+        return data
+
+    def _concate_preds(self, _y, preds):
+        _y = pd.concat([_y, preds])
+        _y = _y.groupby(level=list(range(_y.index.nlevels)))
+        _y = _y.aggregate(self.aggregate_method)
```

Comment:
I am not fully clear why is this aggregation required. Can you please give an example?

Reply:
let me know if this example does not clear it, I explained aggregate at the end - https://github.com/sktime/sktime/pull/6842#issuecomment-2281679977

Reply:
Yes, still unclear, sorry. If we have data from history, why will we aggregate with prediction (with error) instead of using actual observation directly?

Reply:
we are aggregating only the future values
1. we had 4 history points and predicted next 3 points.
2. we use the first point from predicted point as a part of history
3. now we have 5 history points and predict next 3 points.
4. we use the first point from predicted point as a part of history but this 6th point was also predicted in step (1) so which point do we chose? we apply an aggregate method here to create 1 value from multiple points.
5. now we have 6 history points and predict next 3 points.
6. we use the first point from predicted point as a part of history but this 7th point was also predicted in step (1) and step (2) so which point do we chose? we apply an aggregate method here to create 1 value from multiple points.
7. and this goes on .... till we reach the max(fh)